lib/gag.py

```diff
@@ -0,0 +1,33 @@
+import datetime
+import mongoengine
+
+class Gag(mongoengine.Document):
+    """Gag details are stored here"""
+    content_type = mongoengine.StringField(required=True)
+    url = mongoengine.StringField(required=True)
+    created_at = mongoengine.DateTimeField(default=datetime.datetime.now)
+    tags = mongoengine.StringField()
+
+    def save_gags(gags):
+        for gag in gags:
+            existing_gag = Gag.objects(url=gag['url']).first()
```

Comment:
how is ordering being handled here?
You are adding new posts if not present, what if the order is changed?  You will have to maintain order in some way

Reply:
changed the order to use updated_at instead of created_at